src/lib/helpers.ts

```diff
@@ -102,3 +114,38 @@ export const readJsonFromFile = (filepath: string): Record<string, unknown> => {
 
   return maybeJson;
 };
+
+export const isValidPkceCodeVerifier = (verifier: string) => {
+  const PKCE_CHALLENGE_REGEX = /^[A-Za-z0-9\-._~]{43,128}$/;
+  return PKCE_CHALLENGE_REGEX.test(verifier);
+};
+
+export const createPKCEVerifier = () => {
+  const randomBytes = crypto.getRandomValues(new Uint8Array(32));
+  return Buffer.from(randomBytes).toString('base64url');
+};
+
+export const supportedPkceAlgorithms = ['plain', 'S256'] as const;
+
+export const createPKCECodeChallenge = async (
+  verifier: string = createPKCEVerifier(),
+  algorithm: PKCEAlgorithm = 'plain'
+) => {
+  let challenge: string;
+
+  switch (algorithm) {
```

Comment:
@tanettrimas How about adding a `default:` clause for this switch and make it throw?

Reply:
Why do I need a default? it is only used with constant values

Reply:
It's good coding practice to leave no execution branches unchecked, no matter how unlikely you think their execution might be.

Every public method (in this case, "public" means "an exported function in a module that someone else can consume") should validate its input, and throw if it receives an input it's not expected to handle.

I can see that you are already guarding against invalid values and returning an HTTP 400 error response if this happens, and _today_ nothing in the rest of the code will pass an invalid value to this method. The thing is that _tomorrow_ some other developer might decide to reuse this method somewhere else and _not_ guard against invalid values, thus producing an unexpected behaviour and unintentionally introducing a bug.